lib/convection/control/stack.rb

```diff
@@ -586,12 +582,39 @@ def get_events(pages = nil, stack_name = id)
         end
       rescue Aws::CloudFormation::Errors::ValidationError # Stack does not exist
       rescue Aws::EC2::Errors::RequestExpired
-        raise unless @credential_error_count <= @credential_error_max_retries
+        raise if reached_max_credential_retries?
+
+        wait_for_credential_refresh
+
+        reload_cf_client
+        retry
+      end
+
+      def client_options
+        client_options = { region: @region }
+        client_options[:credentials] = @credentials unless @credentials.nil?
 
+        client_options
+      end
+
+      def reload_cf_client
+        Aws.shared_config.fresh if @credentials.nil?
```

Comment:
@athompson-r7 sorry, I was on holiday. Back now!

> This needs to continue to work with IAM user credentials, have you verified that?

I've implemented for compatibility (e.g. an early return if we have set `@credentials`), but I haven't done testing locally.

> Can we create a new Client obj?

Yes, creating a new client object loads credentials from `Aws.shared_config`.

> Why do we need to refresh the credentials?

Otherwise the credentials read at the first client's initialisation will be used.

> Can we let the user refresh the credentials themselves?

It depends:
* Consumers of convection (as library code) **can** refresh credentials themselves.
* Users of the convection binary **must** re-run any commands that were interrupted between requests due to expired credentials.

> It seems like credentials expiring in the middle of the command shouldn't affect the outcome.

The goal of this pull request is to ensure that the request convection sends uses credentials from `~/.aws/credentials` (granted the `:credentials` option was not passed in during initialisation in which case we maintain the previous functionality).

> That is the command should be immutable.

The command itself was previously being interrupted by expired credentials in the middle of multiple operations. This just ensures the latest content of the credentials file is used.